code/mission/import/xwinglib.cpp

```diff
@@ -582,5 +599,161 @@ bool XWingMission::load(XWingMission *m, const char *data)
 		m->flightgroups.push_back(*nfg);
 	}
 
+	for (int n = 0; n < h->number_of_objects; n++) {
+		xwi_objectgroup *oj =
+			(xwi_objectgroup*)(data + sizeof(xwi_header) + sizeof(xwi_flightgroup) + sizeof(xwi_objectgroup) * n);
```

Comment:
The `sizeof(xwi_flightgroup)` should be multiplied by `h->number_of_flight_groups` because the object data starts after *all* the flight groups have been parsed.